asmcli/lib/parse.sh

```diff
@@ -473,3 +473,157 @@ arg_required() {
     fatal "Option ${1} requires an argument."
   fi
 }
+
+x_parse_install_args() {
+  if [[ "${*}" = '' ]]; then
+    x_usage >&2
+    exit 2
+  fi
+
+  # shellcheck disable=SC2064
+  trap "$(shopt -p nocasematch)" RETURN
+  shopt -s nocasematch
+
+  REVISION_LABEL_REGULAR="asm-managed"; readonly REVISION_LABEL_REGULAR
+  REVISION_LABEL_RAPID="asm-managed-rapid"; readonly REVISION_LABEL_RAPID
+  REVISION_LABEL_STABLE="asm-managed-stable"; readonly REVISION_LABEL_STABLE
+  while [[ $# != 0 ]]; do
+    case "${1}" in
+      -l | --cluster_location | --cluster-location)
+        arg_required "${@}"
+        context_set-option "CLUSTER_LOCATION" "${2}"
+        shift 2
+        ;;
+      -n | --cluster_name | --cluster-name)
+        arg_required "${@}"
+        context_set-option "CLUSTER_NAME" "${2}"
+        shift 2
+        ;;
+      --kc | --kubeconfig)
+        arg_required "${@}"
+        context_set-option "KUBECONFIG" "${2}"
+        context_set-option "KUBECONFIG_SUPPLIED" 1
+        shift 2
+        ;;
+      --ctx | --context)
+        arg_required "${@}"
+        context_set-option "CONTEXT" "${2}"
+        shift 2
+        ;;
+      -p | --project_id | --project-id)
+        arg_required "${@}"
+        context_set-option "PROJECT_ID" "${2}"
+        shift 2
+        ;;
+      --fleet_id | --fleet-id)
+        arg_required "${@}"
+        context_set-option "FLEET_ID" "${2}"
+        shift 2
+        ;;
+      -e | --enable_all | --enable-all)
+        context_set-option "ENABLE_ALL" 1
+        shift 1
+        ;;
+      --enable_cluster_roles | --enable-cluster-roles)
+        context_set-option "ENABLE_CLUSTER_ROLES" 1
+        shift 1
+        ;;
```

Comment:
Cluster roles shouldn't be necessary anymore, since the installer isn't getting run as whoever is running the script.

Reply:
Done